### PR TITLE
Search songs, podcasts, albums

### PR DESCRIPTION
**Pull request by acherkashin**
Functionality requested in the issue following issue #3 

- [x] Button to run search action
- [x] Button to clean search action 
- [x] Add Welcome view when no searches
- [x] Display search results 
- [ ] Show the latest user search queries

image

- [ ] We need to expand search explorer if we perform search and explorer is collapsed?
- [ ] Need to open left panel when run search action but left panel is collapsed
- [ ] Need to clean up search from saved playlists

### CODE REVIEW

File: src/tree/searchTree.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    refresh(item?: vscode.TreeItem): void {
+        this._onDidChangeTreeData.fire(item);
+    }
+
+    getTreeItem(element: vscode.TreeItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        return element;
+    }
+
+    async getChildren(element?: vscode.TreeItem) {
```
Comment (acherkashin) [inline src/tree/searchTree.ts:27]: Need to refactor this method

Comment (AlexSnowLeo) [reply src/tree/searchTree.ts:27]: I think we should create issue for this refactoring or TODO marked?

Comment (acherkashin) [reply src/tree/searchTree.ts:27]: I've refactored this code, @AlexSnowLeo  could you look at this maybe you will have an idea how we can improve it and make even better.

File: src/utils/iconUtils.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { PlayList } from "../yandexApi/playlist/playList";
 import { GeneratedPlayList } from "../yandexApi/feed/generatedPlayList";
 import { getExtensionPath } from "./extensionUtils";
 
-export function getThemeIcon(iconFileName: string): ThemeIcon {
```
[the remark this answers is not in the document]
Comment (acherkashin) [reply src/utils/iconUtils.ts:7]: @AlexSnowLeo maybe we need to setup tslint rules to fail when there are such issues

File: package.json [hunk 3]
```diff
@@ -204,9 +213,19 @@
...
+          "id": "yandex-music-search",
+          "name": "Search"
         }
       ]
-    }
+    },
+    "viewsWelcome": [
+      {
+        "view": "yandex-music-search",
+        "contents": "Поиск по трекам, альбомам, исполнителчям и подкастам. \n[Поиск](command:yandexMusic.search)"
```
[the remark this answers is not in the document]
Comment (acherkashin) [reply package.json]: It is fixed in master

**Pull request merged**